gems/aws-sdk-core/lib/aws-sdk-core/plugins/user_agent.rb

```diff
@@ -48,15 +48,13 @@ def self.feature(_feature, &block)
         block.call
       end
 
-      def self.metric(metric, &block)
+      def self.metric(*metrics, &block)
         Thread.current[:aws_sdk_core_user_agent_metric] ||= []
-        Thread.current[:aws_sdk_core_user_agent_metric] << METRICS[metric]
+        metrics = metrics.map { |metric| METRICS[metric] }
```

Comment:
I think we may want to do a `compact` here, this lets us avoid creating an array and passing it and we can do something like:
```ruby
Aws::Plugins::UserAgent.metric((METRIC_1 if condition1), (METRIC_2 if condition2)) do

end
```